### PR TITLE
Prevent cluster deletion hangs by using heartbeat condition

### PR DESCRIPTION
**Pull request by csrwng**
**What this PR does / why we need it**:
If the hcco is not available, the cpo will wait indefinitely for it to update hcp conditions to indicate that guest resources have been deleted. To solve that, the hcco will periodically update a condition to let the CPO that it's working on deletion. If the condition is not updated after a certain amount of time, the CPO gives up waiting and proceeds with the rest of the deletion.

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes # [HOSTEDCP-667](https://issues.redhat.com//browse/HOSTEDCP-667)

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.

### CODE REVIEW

File: control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go [hunk 1]
```diff
@@ -3217,13 +3220,29 @@ func (r *HostedControlPlaneReconciler) removeCloudResources(ctx context.Context,
 	log.Info("Removing cloud resources")
 
 	// check if resources have been destroyed
-	if meta.IsStatusConditionTrue(hcp.Status.Conditions, string(hyperv1.CloudResourcesDestroyed)) {
+	resourcesDestroyedCond := meta.FindStatusCondition(hcp.Status.Conditions, string(hyperv1.CloudResourcesDestroyed))
+	if resourcesDestroyedCond != nil && resourcesDestroyedCond.Status == metav1.ConditionTrue {
```
Comment (enxebre) [inline control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go:3224]: isn't `resourcesDestroyedCond != nil && resourcesDestroyedCond.Status == metav1.ConditionTrue`
The exact underlying impl of `IsStatusConditionTrue` (which calls `IsStatusConditionPresentAndEqual`)? why the change?

File: control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go [hunk 2]
```diff
@@ -3217,13 +3220,29 @@ func (r *HostedControlPlaneReconciler) removeCloudResources(ctx context.Context,
...
+
+		// Determine if too much time has passed since the last time we got an update
+		var timeElapsed time.Duration
+		if resourcesDestroyedCond != nil {
+			timeElapsed = time.Since(resourcesDestroyedCond.LastTransitionTime.Time)
+		} else {
+			timeElapsed = time.Since(cvoScaledDownCond.LastTransitionTime.Time)
+		}
+
+		if timeElapsed > resourceDeletionTimeout {
```
Comment (enxebre) [inline control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go:3242]: Can we add a code comment here for readers with less context to understand where we are coming from when getting here? e.g.
```
// Best effort heuristic to decided the HCCO is not able to operate.
timeElapsed > resourceDeletionTimeout
```

I'm also not fully sold on skipping here silently potentially leaving cloud resources behind, but I can't propose anything better atm.

File: control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go (hunk 1 above)
Comment (csrwng) [reply control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go:3224]: The only reason is that it's used below (line 3236), but we can re-fetch if that looks better.

**Comment by enxebre** [comment]
It's fairly easy to end up in this scenario if you e.g. hit quotas on the LB management side. I think we can move with this for now and iterate based on usage feedback.
/lgtm
